caravel/assets/visualizations/nvd3_vis.js

```diff
@@ -67,8 +67,23 @@ function nvd3Vis(slice) {
         return;
       }
 
+      const axisLabels = payload.data[0].values;
+      let stretchMargin = 0;
+      // Calculates the longest label size for stretching bottom margin
+      function adjustMargins() {
```

Comment:
Since heatmap and bar plot have different data format (heatmap has a 2d data array and bar plot's data are all stored as 1d in data[0].values, there wasn't a nice way to make them use the same adjustMargin() function.